core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -288,4 +288,39 @@ object FileUtils {
     }
     return null
   }
+
+  private fun getDecodedFileName(url: String?, src: String?): String =
```

Comment:
Can we have a comment explaining does this function? It should be as well unit trsted. Please use standart library to parse URL and URL decode it. Here the implementation looks like really weak.

Reply:
@kelson42 I created this function. I should implement the docs and tests.